### PR TITLE
Retry rewind if doc and update versions don't match

### PR DESCRIPTION
**Pull request by jpallen**

### CODE REVIEW

File: app/coffee/DiffManager.coffee [hunk 1]
```diff
@@ -49,7 +69,9 @@ module.exports = DiffManager =
 
 			lastUpdate = updates[0]
 			if lastUpdate? and lastUpdate.v != version - 1
-				return callback new Error("latest update version, #{lastUpdate.v}, does not match doc version, #{version}")
+				error = new Error("latest update version, #{lastUpdate.v}, does not match doc version, #{version}")
+				error.retry = true
+				callback error
```
Comment (briangough) [inline app/coffee/DiffManager.coffee]: missing a **return** on the callback(error) here

File: app/coffee/DiffManager.coffee [hunk 2]
```diff
@@ -34,7 +33,28 @@ module.exports = DiffManager =
...
+		# Whichever order we get the latest document and the latest updates,
+		# there is potential for updates to be applied between them so that
+		# they do not return the same 'latest' versions.
+		# If this happens, we just retry and hopefully get them at the compatible
+		# versions.
+		retries = 3
+		callback = (error, args...) ->
+			if error?
+				if error.retry and retries > 0
+					logger.warn {error, project_id, doc_id, version}, "retrying getDocumentBeforeVersion"
```
Comment (briangough) [inline app/coffee/DiffManager.coffee]: could be useful to log the number of retries in the warn

File: app/coffee/DiffManager.coffee (hunk 1 above)
Comment (jpallen) [inline app/coffee/DiffManager.coffee]: Good catch

File: app/coffee/DiffManager.coffee (hunk 2 above)
Comment (jpallen) [inline app/coffee/DiffManager.coffee]: Done

**Pull request merged**